SEBrowser/Scripts/TSX/Components/SettingsSlice.ts

```diff
@@ -22,31 +22,54 @@
 //       Cleaned up Settings code.
 //
 //******************************************************************************************************
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import { Redux, SEBrowser } from '../global';
 
-const settingsSlice = createSlice({ 
+declare var homePath: string;
+
+export const LoadSettings = createAsyncThunk('Settings/LoadSettingsThunk', async () => {
+    return loadTimeZone();
+});
+
+
+const settingsSlice = createSlice({
     name: 'Settings',
 
     initialState: {
-        eventSearch: { NumberResults: 100 }
+        eventSearch: { NumberResults: 100 },
+        TimeZoneOffset: 0,
     } as Redux.SettingsState,
-    
+
     reducers: {
-        LoadSettings: (state) => {
+        SetEventSearch: (state: Redux.SettingsState, action: { type: string, payload: Redux.IEventSearchSettings }) => {
+            state.eventSearch = action.payload;
+            saveSettings(state);
+        },
+    },
+    extraReducers: (builder) => {
+
+        builder.addCase(LoadSettings.fulfilled, (state, action) => {
             let preserved = readSettings();
+
             if (preserved != undefined) {
                 state.eventSearch = preserved.eventSearch;
             }
-            else {
+            else
                 state.eventSearch = { NumberResults: 100 };
+
+            state.TimeZoneOffset = action.payload;
+        });    
+        
+        builder.addCase(LoadSettings.rejected, (state, action) => {
+            let preserved = readSettings();
+
+            if (preserved != undefined) {
+                state.eventSearch = preserved.eventSearch;
             }
-            return state
-        },
-        SetEventSearch: (state: Redux.SettingsState, action: { type: string, payload: Redux.IEventSearchSettings }) => {
-            state.eventSearch = action.payload;
-            saveSettings(state);
-        },
+            else
+                state.eventSearch = { NumberResults: 100 };
+            state.TimeZoneOffset = 0;
```

Comment:
The `else` block only applies if `preserved` is `undefined` where `preserve` is the state from local storage. Basically it uses `100` if the preserved state is not available.
`TimeZoneOffset` comes from the Server I don't think we want to preserve that one since it is the difference between UTC and systemTime which is not a per user setting but a global setting. So preserving that would lead to caching issues.